pkg/hub/submarineragent/controller.go

```diff
@@ -202,7 +198,7 @@ func (c *submarinerAgentController) sync(ctx context.Context, syncCtx factory.Sy
 			return err
 		}
 
-		config, err := c.configLister.SubmarinerConfigs(name).Get(helpers.SubmarinerConfigName)
+		config, err := c.configLister.SubmarinerConfigs(namespace).Get(name)
```

Comment:
This is incorrect.  We get here b/c the `ManagedCluster`, `ManifestWork` or `ManagedClusterAddOn` changed, not the `SubmarinerConfig`, and `namespace` is empty so this lookup will always return `NotFound`.